research/deep_speech/deep_speech.py

```diff
@@ -0,0 +1,269 @@
+# Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Deep Speech
+"""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+# pylint: disable=g-bad-import-order
+import numpy as np
+from absl import app as absl_app
+from absl import flags
+import tensorflow as tf
+# pylint: enable=g-bad-import-order
+
+import deep_speech_model
+import data.dataset as dataset
+from official.utils.export import export
+from official.utils.flags import core as flags_core
+from official.utils.logs import hooks_helper
+from official.utils.logs import logger
+from official.utils.misc import distribution_utils
+from official.utils.misc import model_helpers
+
+
+def convert_keras_to_estimator(keras_model, num_gpus, flags_obj):
+  """Configure and convert keras model to Estimator.
+  Args:
+    keras_model: A Keras model object.
+    num_gpus: An integer, the number of gpus.
+    flags_obj: An object containing parsed flags. See define_resnet_flags()
+      for details.
+  Returns:
+    estimator: The converted Estimator.
+  """
+  optimizer = tf.keras.optimizers.SGD(
+      lr=flags_obj.learning_rate, momentum=flags_obj.momentum,
+      decay=flags_obj.l2, nesterov=True)
+  keras_model.compile(
+      optimizer=optimizer, loss={"ctc_loss": lambda y_true, y_pred: y_pred})
+
+  distribution_strategy = distribution_utils.get_distribution_strategy(
+      num_gpus)
+  run_config = tf.estimator.RunConfig(
+      train_distribute=distribution_strategy, session_config=session_config)
+
+  estimator = tf.keras.estimator.model_to_estimator(
+      keras_model=keras_model, model_dir=flags_obj.model_dir, config=run_config)
+
+  return estimator
+
+
+def generate_dataset(data_dir):
+  """Generate a speech dataset."""
+  audio_conf = dataset.AudioConfig(
+      flags_obj.sample_rate, flags_obj.frame_length, flags_obj.frame_step)
+  train_data_conf = dataset.DatasetConfig(
+      audio_conf,
+      data_dir,
+      flags_obj.vocabulary_file,
+  )
+  speech_dataset = dataset.DeepSpeechDataset(train_data_conf)
+  return speech_dataset
+
+
+def main(_):
+  with logger.benchmark_context(flags_obj):
+    run_deep_speech(flags_obj)
+
+
+def run_deep_speech(_):
```

Comment:
Since `main` is calling `run_deep_speach(flags_obj)`, `_` should be changed to `flags_obj` here.